### DISCUSSION TITLE
Pull requests messages seem to blame user rather than Github in certain cases

### DISCUSSION THREAD

**Discussion by pdewouters** [community/community#41916] [Pull Requests]
### Select Topic Area

Product Feedback

### Body

This is something that should be changed IMO, in this specific case, the repo was setup by another developer, the main branch was not protected and pull requests were set to delete the branch automatically after merging. 

So in this particular case, I wanted to sync a staging branch with the main branch, and because I did not have the ability to push from my local, I made a PR main -> stage. When I merged this, it deleted the main branch which is the deployable branch. Another developer noticed this days later, and I got the blame because that is what it looks like, the text is "pdewouters deleted the branch", although it was the repo settings that caused this. Regardless of blame, or how the repo was setup incorrectly, it is a bad experience for the developer, and this could lead to serious consequences for the developer.

I would suggest:
- changing the wording of the message when the branch is deleted, if it is not a manual deletion by a human
- some sort of confirmation, if this is a long standing branch or if the name is main or master.
- notifications to the repo members (maybe that's already the case)

**Comment by jfullstackdev** [comment]
why would you be needing to delete the branch right away ? 

in my practice, even for the company, I'm just using a branch a single time when I'm doing a certain task, after that it's discarded but it's clean because, say

`december17-sprint1`
`december17-sprint2` 
`december17-sprint3`

I believe branches are just single use, because, it can be reverted easily once you push something on the production that is not working

as you can see above, it can be my entire day working but it's counting, so I don't reuse branches because they are intended for single use only
